File: backend/src/main/java/todoktodok/backend/book/application/dto/response/AladinBookResponse.java
package todoktodok.backend.book.application.dto.response;

import todoktodok.backend.book.infrastructure.aladin.AladinItemResponse;

public record AladinBookResponse(
        String bookId,
        String bookTitle,
        String bookAuthor,
        String bookImage
) {

    public AladinBookResponse(AladinItemResponse aladinItemResponse) {
        this(
                aladinItemResponse.isbn13(),
                aladinItemResponse.title(),
                aladinItemResponse.author(),
                aladinItemResponse.cover()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/book/AladinBookResponse.kt
package com.team.todoktodok.data.network.response.book

import com.team.domain.model.book.AladinBook.Companion.AladinBook
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AladinBookResponse(
    @SerialName("bookId")
    val isbn: Long,
    @SerialName("bookTitle")
    val title: String,
    @SerialName("bookAuthor")
    val author: String,
    @SerialName("bookImage")
    val image: String,
)

fun AladinBookResponse.toDomain() =
    AladinBook(
        isbn = isbn,
        title = title,
        author = author,
        image = image,
    )
